walkthroughs/howto-tls-file-provided/README.md

```diff
@@ -0,0 +1,331 @@
+
+```bash
+aws ec2 create-key-pair --key-name color-app | jq -r .KeyMaterial > ~/.ssh/color-app.pem
+```
+
+This command creates an Amazon EC2 Key Pair with name `color-app` and saves the private key at `~/.ssh/color-app.pem`.
+
+Next, we need to set a few environment variables before provisioning the
+infrastructure. Please change the value for `AWS_ACCOUNT_ID`, `KEY_PAIR_NAME`, and `ENVOY_IMAGE` below.
+
+```bash
+export AWS_ACCOUNT_ID=<your account id>
+export KEY_PAIR_NAME=<color-app or your SSH key pair stored in AWS>
+export AWS_DEFAULT_REGION=us-west-2
+export ENVIRONMENT_NAME=AppMeshTLSExample
+export MESH_NAME=ColorApp-TLS
+export AWS_ENVOY_IMAGE=<get the latest from https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html>
+export SERVICES_DOMAIN="default.svc.cluster.local"
+export GATEWAY_IMAGE_NAME="gateway"
+export COLOR_TELLER_IMAGE_NAME="colorteller"
+```
+
+First, create the VPC.
+
+```bash
+./infrastructure/vpc.sh
+```
+
+Next, create the ECS cluster and ECR repositories.
+
+```bash
+./infrastructure/ecs-cluster.sh
+./infrastructure/ecr-repositories.sh
+```
+
+Next, build and deploy the color app images.
+
+```bash
+./src/colorteller/deploy.sh
+./src/gateway/deploy.sh
+```
+
+## Step 3: Generate the Certficates
+
+Before we can encrypt traffic between services in the mesh, we need to generate our certificates.
+
+For this demo, we are going to set up two separate Certificate Authorities. The first one will be used to sign the certificate for the White Color Teller, the second will be used to sign the certificate for the Green Color Teller.
+
+```bash
+./src/tlsCertificates/certs.sh
+```
+
+This generates a few different files
+
+- *_cert.pem: These files are the public side of the certificates
+- *_key.pem: These files are the private key for the certificates
+- *_cert_chain: These files are an ordered list of the public certificates used to sign a private key
+- ca_1_ca_2_bundle.pem: This file contains the public certificates for both CAs.
+
+You can verify that the White Color Teller certificate was signed by CA 1 using this command.
+
+```bash
+openssl verify -verbose -CAfile src/tlsCertificates/ca_1_cert.pem  src/tlsCertificates/colorteller_white_cert.pem
+```
+
+We are going to store these certificates in [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/). This will allow us to securely retrieve them at a later time.
+
+```bash
+./src/tlsCertificates/deploy.sh
+```
+
+## Step 4: Export our Custom Docker Image
+
+Finally, we can build and deploy our custom docker image. This container has a `/keys` directory and a custom startup script that will pull the necessary certificates from `AWS Secrets Manager`.
+
+```bash
+./src/customEnvoyImage/deploy.sh
+```
+
+The script will print the image location in ECR. Export it as:
+
+```bash
+export ENVOY_IMAGE=<output of last command>
```

Comment:
Total nitpick: we end up calling this `colorteller-envoy`, but it's also used for gateway. How about something more generic? Like `secure-envoy` or just `colorapp-envoy`?